### PR TITLE
Fix deleting sub menus

### LINKED ISSUE #16989: Unable to delete sub menus
8 months ago we fixed an issue with deleting the menu #15500 but it seems no one reported that sub-menus are not deleted even after the fix. This PR fixes both

Related to #15497

/cc @sarahelsaig

### PR DESCRIPTION
**Pull request by hishamco**
Fixes #16989

### CODE REVIEW

**Comment by hishamco** [comment]
Tried & tested on `TheBlogTheme`

File: src/OrchardCore.Modules/OrchardCore.Menu/Controllers/AdminController.cs [hunk 1]
```diff
@@ -271,8 +271,9 @@ public async Task<IActionResult> Delete(string menuContentItemId, string menuIte
             return NotFound();
         }
 
-        var menuItems = menuContentAsJson[nameof(MenuItemsListPart)]?[nameof(MenuItemsListPart.MenuItems)] as JsonArray;
-        menuItems?.Remove(menuItem);
+        var menuItems = menuContentAsJson.SelectNode(menuItem.Parent.GetPath()) as JsonArray;
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.Menu/Controllers/AdminController.cs:274]: Count Parent here be null like if you remove the last parent?

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Menu/Controllers/AdminController.cs:274]: I already that about that, all the menu items have a parent, let me double-check and delete all the menus in the blog theme

Comment (hishamco) [reply src/OrchardCore.Modules/OrchardCore.Menu/Controllers/AdminController.cs:274]: I deleted all the menus in the main menu, it works like a charm

**Review by MikeAlhayek — APPROVED** [review]

**Pull request merged**